backend/pkg/database/odatasql/query.go

```diff
@@ -592,19 +611,107 @@ func buildWhereFromFilter(sqlVariant jsonsql.Variant, schemaMetas map[string]Sch
 			r := strings.ReplaceAll(right, "'", "")
 			value = fmt.Sprintf(sqlOperators[operator], r)
 		default:
-			return query, fmt.Errorf("unsupported token type")
+			return query, fmt.Errorf("unsupported token type %s", node.Children[1].Token.Type)
 		}
+
 		query = fmt.Sprintf(
 			"%s LIKE '%s'",
 			sqlVariant.JSONExtractText(fieldSource, fmt.Sprintf("$.%s", queryPath)),
 			value,
 		)
+	default:
+		return query, fmt.Errorf("unsupported operator: %s", operator)
 	}
 
 	return query, nil
 }
 
+// nolint:cyclop
+func fieldMetaFromQueryPath(schemaMetas map[string]SchemaMeta, field FieldMeta, path string) ([]FieldMeta, error) {
+	switch field.FieldType {
+	case PrimitiveFieldType:
+		if path == "" {
+			return []FieldMeta{field}, nil
+		} else {
+			return nil, fmt.Errorf("can not subpath a primitive type")
+		}
+	case CollectionFieldType:
+		return fieldMetaFromQueryPath(schemaMetas, *field.CollectionItemMeta, path)
```

Comment:
This is a recursive algorithm for walking the schema tree, I'm sure we could implement a non-recursive version of it but I expect it would be more complex and harder to understand and involve some messy loops.